### PR TITLE
Testing lib

### LINKED ISSUE #12: Add form validations
- Use react testing library
- Test your form validations (border color red?, error message rendered or not)

### PR DESCRIPTION
**Pull request by borangoher**
close #12. This PR should now display only the relevant changes.

### CODE REVIEW

File: src/Components/payment-form/__tests__/paymentForm.test.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+  await waitFor(() => {
+    expect(
+      screen.getByText(/payment.errorMessages.cardholderNameMin/i)
+    ).toBeInTheDocument();
+  });
+
+  expect(
+    screen.getByText(/payment.errorMessages.cardholderNameMin/i)
+  ).toBeInTheDocument();
```
Comment (yeged) [inline src/Components/payment-form/__tests__/paymentForm.test.js:67]: is it duplicated
`expect(
    screen.getByText(/payment.errorMessages.cardholderNameMin/i)
  ).toBeInTheDocument();`

**Review by yeged — APPROVED** [review]

File: src/Components/login/__tests__/Login.test.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+};
+
+it("should log user in for a valid input", async () => {
+  render(<MockLogin />);
+
+  const usernameElement = screen.getByLabelText(/login.username/i);
+  const passwordElement = screen.getByLabelText(/login.password/i);
+  const loginButton = screen.getByRole("button", { value: /login.login/i });
+
+  fireEvent.change(usernameElement, { target: { value: "abcde" } });
```
Comment (ozgunbal) [inline src/Components/login/__tests__/Login.test.js:25]: try to use https://testing-library.com/docs/user-event/intro/ . This is better for simulating user's behaviors. fireEvent assumes some internal logic. Black box testing is more beneficial in our cases.

File: src/Components/login/__tests__/Login.test.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+const MockLogin = () => {
+  return (
+    <BrowserRouter>
+      <LoginProvider>
+        <Login />
+      </LoginProvider>
+    </BrowserRouter>
+  );
+};
+
```
Comment (ozgunbal) [inline src/Components/login/__tests__/Login.test.js:17]: - Recommending to read this article: https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#adding-aria--role-and-other-accessibility-attributes-incorrectly
- After handling failed login in #25, add a test for un-auth entry.(send 401 from msw for test case, then expect to see error message in the document)

**Review by ozgunbal — CHANGES_REQUESTED** [review]

**Review by ozgunbal — APPROVED** [review]
Approved but expecting to finish #27 in a different PR.

**Pull request merged**